src/transformers/models/dallemega/modeling_dallemega.py

```
@@ -0,0 +1,1557 @@
+                returned tensors for more detail.
+            output_hidden_states (`bool`, *optional*):
+                Whether or not to return the hidden states of all layers. See `hidden_states` under returned tensors
+                for more detail.
+            return_dict (`bool`, *optional*):
+                Whether or not to return a [`~utils.ModelOutput`] instead of a plain tuple.
+        """
+        output_attentions = output_attentions if output_attentions is not None else self.config.output_attentions
+        output_hidden_states = (
+            output_hidden_states if output_hidden_states is not None else self.config.output_hidden_states
+        )
+        use_cache = use_cache if use_cache is not None else self.config.use_cache
+        return_dict = return_dict if return_dict is not None else self.config.use_return_dict
+        superconditioning_scale = superconditioning_scale if superconditioning_scale is not None else self.config.superconditioning_scale
+
+        do_superconditioning = superconditioning_scale > 1 and not self.training
+
+        # retrieve input_ids and inputs_embeds
+        if input_ids is not None and inputs_embeds is not None:
+            raise ValueError("You cannot specify both decoder_input_ids and decoder_inputs_embeds at the same time")
+        elif input_ids is not None:
+            input_shape = input_ids.size()
+            input_ids = input_ids.view(-1, input_shape[-1])
+        elif inputs_embeds is not None:
+            input_shape = inputs_embeds.size()[:-1]
+        else:
+            raise ValueError("You have to specify either decoder_input_ids or decoder_inputs_embeds")
+
+        if do_superconditioning:
+            encoder_hidden_states = torch.cat([encoder_hidden_states, encoder_hidden_states_uncond], dim=0)
+            input_ids = input_ids.repeat(2, 1)
+            attention_mask = attention_mask.repeat(2, 1)
+        
```

The `encoder_hidden_states_uncond` will be passed from. `DalleMegaModel`. We concatenate those two here
and repeat the decoder inputs.

---

Cool! Maybe we can also raise a nice error message if the `encoder_hidden_states_uncond` are in the wrong format or None but `do_superconditioning` is True?